romancal/tweakreg/tests/test_tweakreg.py

```diff
@@ -1185,3 +991,80 @@ def test_parse_catfile_raises_error_on_invalid_content(tmp_path, catfile_line_co
+
+
+@pytest.mark.parametrize(
+    "exposure_type",
+    ["WFI_GRISM", "WFI_PRISM", "WFI_DARK", "WFI_FLAT", "WFI_WFSC"],
+)
+def test_tweakreg_skips_invalid_exposure_types(exposure_type, tmp_path, base_image):
+    """Test that TweakReg updates meta.cal_step with tweakreg = COMPLETE."""
+    img1 = base_image(shift_1=1000, shift_2=1000)
+    img1.meta.exposure.type = exposure_type
+    img2 = base_image(shift_1=1000, shift_2=1000)
+    img2.meta.exposure.type = exposure_type
+    res = trs.TweakRegStep.call([img1, img2])
+
+    assert type(res) == ModelLibrary
+    with res:
+        for i, model in enumerate(res):
+            assert hasattr(model.meta.cal_step, "tweakreg")
+            assert model.meta.cal_step.tweakreg == "SKIPPED"
+            res.shelve(model, i, modify=False)
+
+
+@pytest.mark.parametrize(
+    "catalog_data, expected_colnames, raises_exception",
+    [
+        # both 'x' and 'y' columns present
+        ({"x": [1, 2, 3], "y": [4, 5, 6]}, ["x", "y"], False),
+        # 'xcentroid' and 'ycentroid' columns present, should be renamed
+        ({"xcentroid": [1, 2, 3], "ycentroid": [4, 5, 6]}, ["x", "y"], False),
+        # 'x' present, 'ycentroid' present, should rename 'ycentroid' to 'y'
+        ({"x": [1, 2, 3], "ycentroid": [4, 5, 6]}, ["x", "y"], False),
+        # 'xcentroid' present, 'y' present, should rename 'xcentroid' to 'x'
+        ({"xcentroid": [1, 2, 3], "y": [4, 5, 6]}, ["x", "y"], False),
+        # neither 'x' nor 'xcentroid' present
+        ({"y": [4, 5, 6]}, None, True),
+        # neither 'y' nor 'ycentroid' present
+        ({"x": [1, 2, 3]}, None, True),
+        # no relevant columns present
+        (
+            {"a": [1, 2, 3], "b": [4, 5, 6]},
+            None,
+            True,
+        ),
+    ],
+)
+def test_validate_catalog_columns(catalog_data, expected_colnames, raises_exception):
+    """Test that TweakRegStep._validate_catalog_columns() correctly validates the
+    presence of required columns ('x' and 'y') in the provided catalog."""
+    catalog = table.Table(catalog_data)
+    if raises_exception:
+        with pytest.raises(ValueError):
+            _validate_catalog_columns(catalog)
+    else:
+        _validate_catalog_columns(catalog)
+        assert set(catalog.colnames) == set(expected_colnames)
+
+
+def test_tweakreg_handles_mixed_exposure_types(tmp_path, base_image):
+    """Test that TweakReg can handle mixed exposure types
+    (non-WFI_IMAGE data will be marked as SKIPPED only and won't be processed)."""
+    img1 = base_image(shift_1=1000, shift_2=1000)
+    add_tweakreg_catalog_attribute(tmp_path, img1, catalog_filename="img1")
+    img1.meta.exposure.type = "WFI_IMAGE"
+
+    img2 = base_image(shift_1=1000, shift_2=1000)
+    add_tweakreg_catalog_attribute(tmp_path, img2, catalog_filename="img2")
+    img2.meta.exposure.type = "WFI_IMAGE"
+
+    img3 = base_image(shift_1=1000, shift_2=1000)
+    img3.meta.exposure.type = "WFI_GRISM"
```

Comment:
```suggestion
    img2 = base_image(shift_1=1000, shift_2=1000)
    img2.meta.exposure.type = "WFI_GRISM"

    img3 = base_image(shift_1=1000, shift_2=1000)
    add_tweakreg_catalog_attribute(tmp_path, img3, catalog_filename="img3")
    img3.meta.exposure.type = "WFI_IMAGE"
```

Thanks! I think the previous bug might not have been caught by this test if the `WFI_GRISM` exposure was last. I added 2 suggestions here to move the `WFI_GRISM` exposure to second-to-last. The test (should) pass both as it is now and with the suggestion.